webhook/certificates/certificates.go

```diff
@@ -67,8 +75,19 @@ func (r *reconciler) reconcileCertificate(ctx context.Context) error {
 	} else if _, haskey := secret.Data[certresources.CACert]; !haskey {
 		logger.Infof("Certificate secret %q is missing key %q", r.secretName, certresources.CACert)
 	} else {
-		// It has all of the keys, it's good.
-		return nil
+		// Check the expiration date of the certificate to see if it needs to be updated
+		cert, err := tls.X509KeyPair(secret.Data[certresources.ServerCert], secret.Data[certresources.ServerKey])
+		if err != nil {
+			logger.Errorf("error creating pem from certificate and key: %v", err)
+		} else {
+			certData, err := x509.ParseCertificate(cert.Certificate[0])
+			if err != nil {
+				logger.Errorf("error parsing certificate: %v", err)
+			}
+			if certData.NotAfter.Sub(time.Now()) > oneWeek {
```

Comment:
What about `time.Since(certData.NotAfter)`? Might need a `-oneWeek`.

Reply:
I think that makes things less readable.  If I were going to rewrite this I'd probably put:
```
if time.Now().Add(oneWeek) < certData.NotAfter
```

Reply:
Should I change this, or leave it how it is?

Reply:
I'd go with Matt's `NotAfter` is always hard to understand.